src/goto-programs/class_hierarchy.h

```diff
@@ -59,4 +60,36 @@ class class_hierarchyt
   void get_parents_trans_rec(const irep_idt &, idst &) const;
 };
 
+/// Class hierarchy graph node: simply contains a class identifier.
+class class_hierarchy_graph_nodet : public graph_nodet<empty_edget>
+{
+ public:
+  /// Class ID for this node
+  irep_idt class_identifier;
+};
+
+/// Class hierarchy, represented using grapht and therefore suitable for use
+/// with generic graph algorithms.
+class class_hierarchy_grapht : public grapht<class_hierarchy_graph_nodet>
+{
+ public:
+  /// Maps class identifiers onto node indices
+  typedef
+    std::unordered_map<irep_idt, node_indext, irep_id_hash> nodes_by_namet;
+
+ private:
+  /// Maps class identifiers onto node indices
+  nodes_by_namet nodes_by_name;
+
+ public:
```

Comment:
My apologies, the key comments had gotten lost: weird 1-char indent and please merge the `public` sections, possibly by moving the `private` part to the bottom.